openwhisk/tar.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package openwhisk
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func openTar(src []byte) (*tar.Reader, error) {
+	// Create a new bytes.Reader from the input byte slice
+	reader := bytes.NewReader(src)
+
+	// Create a new gzip.Reader from the bytes.Reader
+	gzipReader, err := gzip.NewReader(reader)
+	if err != nil {
+		return nil, err
+	}
+	defer gzipReader.Close()
+
+	// Create a new tar.Reader from the gzip.Reader
+	tarReader := tar.NewReader(gzipReader)
+
+	return tarReader, nil
+}
+
+func UnTar(src []byte, dest string) error {
+	r, err := openTar(src)
+	if err != nil {
+		return err
+	}
+	Debug("open Tar")
+	os.MkdirAll(dest, 0755)
+	for {
+		header, err := r.Next()
+		fmt.Println("Err", err)
```

Comment:
Left over debugging code?  Not sure why we want a println here.

Reply:
Yes that was left from debugging has been removed now @dgrove-oss